### ISSUE TITLE
Incorrect "expected at least one HTML element tag inside a component, but the component is empty" in 0.14.0-pre

### ISSUE THREAD

**Issue by henrik** [phoenixframework/phoenix_live_view#1035]
In the process of making a repro for #1034, I ran into this – don't know if it helps explain #1034 or not, but it seems to at least be a bug in its own right.

This is on LiveView 0.14.0-pre, commit dab960c. I have tried removing node_modules and reinstalling them.

I get this JS error logged:

```
phoenix_live_view.js:1 expected at least one HTML element tag inside a component, but the component is empty:
 <div id="item-1" data-phx-component="2">
    <p>hi</p>
  </div>
````

But as you will note, the component isn't empty.

Repro follows. Issue goes away if I remove the intermediate ListComponent.

``` elixir
defmodule MyItem do
  defstruct id: 0
end

defmodule RemitWeb.DebugLive do
  use Phoenix.LiveView

  @impl true
  def render(assigns) do
    ~L"""
    <%= live_component @socket, ListComponent, id: :my_id %>
    """
  end

  @impl true
  def mount(_params, _session, socket) do
    {:ok, socket}
  end
end

defmodule ListComponent do
  use Phoenix.LiveComponent

  @impl true
  def render(assigns) do
    ~L"""
    <%= for item <- @items do %>
      <%= live_component @socket, ItemComponent, id: item.id %>
    <% end %>
    """
  end

  @impl true
  def mount(socket) do
    socket = assign(socket, items: [
      %MyItem{id: 1},
    ])

    {:ok, socket}
  end
end

defmodule ItemComponent do
  use Phoenix.LiveComponent

  @impl true
  def render(assigns) do
    ~L"""
      <div id="item-<%= @id %>">
        <p>hi</p>
      </div>
    """
  end
end
```

**Comment by henrik** [comment]
On page load I see a brief flash of content, and the content is there in the HTML source.

**Comment by chrismccord** [comment]
what is happening is we are requiring at least one HTML tag as part of the component itself, but in this case the component renders nothing but other components, so in effect there is no identifying tag in the DOM for which to scope this component to. It may be possible to make this work, but I'm not sure because I believe your `ListComponent` should contain a top-level `ul` or `div` which renders the children. I think the error could be improved in this particular case. Does that make sense? Thanks!

**Comment by henrik** [comment]
Ah! Yeah, that makes sense.

Would it be feasible to include the component name and/or ID in the error? Here's a first stab at an improved message:

> expected at least one HTML element tag inside a component, but ListComponent :my_id doesn't contain any elements directly (even though it may have subcomponents that do)

Not sure if that parenthesised bit is necessary, but might make things clearer.

If the specific component is named, I'm not sure if showing its contents (as with the current error) is necessary. Wouldn't it either be empty (in which case there's nothing to output) or it shows the contents of child components, which may confuse more than they help?

**Comment by josevalim** [comment]
Unfortunately the message comes from the client which does not have the component name and ID information. :/ but maybe we can improve it a bit indeed.

**Comment by chrismccord** [comment]
Closed via 64e75b24e3f9828df68c4d10ea11fe74f2727160 . Thanks!

**Comment by henrik** [comment]
@chrismccord Good job! That’s much clearer 👍🏻
